### PR TITLE
Make ToolInfo methods non-static

### PR DESCRIPTION
**Pull request by julienfalque**
Extracted from #3195.

### CODE REVIEW

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -150,10 +155,12 @@
     public function __construct(
         ConfigInterface $config,
         array $options,
-        $cwd
+        $cwd,
+        ToolInfo $toolInfo
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: add missing `@param` for constructor

File: src/ToolInfo.php [hunk 2]
```diff
@@ -27,73 +27,62 @@
 
     const COMPOSER_LEGACY_PACKAGE_NAME = 'fabpot/php-cs-fixer';
 
-    public static function getComposerInstallationDetails()
-    {
-        static $result;
+    private $composerInstallationDetails;
+
+    private $isInstalledByComposer;
```
Comment (keradus) [inline src/ToolInfo.php]: typehints are missing

File: src/Console/ConfigurationResolver.php (hunk 1 above)
Comment (julienfalque) [reply src/Console/ConfigurationResolver.php]: This is conflicting with the config of the project. Shall I ajust `phpdoc_add_missing_param_annotation` to set `only_untyped` to `false`?

Comment (keradus) [reply src/Console/ConfigurationResolver.php]: is it conflicting? `$config` is in the docblock already

Comment (julienfalque) [reply src/Console/ConfigurationResolver.php]: You asked me to add a missing _typed_ `@param` while in our config we don't want to automatically add those.

Comment (keradus) [reply src/Console/ConfigurationResolver.php]: then I see conflict, as `@param $toolInfo` was not required, and `@param $config` is already there...

**Comment by keradus** [comment]
In PR that is origin for this PR, we actually have `ToolInfoInterface`, we shall have it here as well IMO

**Comment by keradus** [comment]
all the rest looks really nice :+1:

**Comment by keradus** [comment]
let m e begin the merging hell...

**Comment by keradus** [comment]
Thank you @julienfalque.

**Pull request closed without merging**